### ISSUE TITLE
Weeping Angels causing massive lag

### ISSUE THREAD

**Issue by EggDoc** [Jeryn99/Weeping-Angels#403]
It all started when I spawned one in my Aternos server to mess around with. And after 1 minute of everything not responding, I got a timed out message. so I tried spawning one in my single-player. It showed me an 882ms server ping.
![image](https://github.com/user-attachments/assets/35d0d8a2-6448-4a38-b6c6-bf4df72de09a)
I don't know if it might be conflicting with other mods so here's the modlist.
![image](https://github.com/user-attachments/assets/d792b66e-b7dd-49c1-b2a2-5e05de43049d)
![image](https://github.com/user-attachments/assets/9705f5bc-1f0f-4b50-ba1a-5badb26d4728)

**Comment by EggDoc** [comment]
here's the log
[latest.log](https://github.com/user-attachments/files/18282629/latest.log)

**Comment by d-umDuck** [comment]
nah, this is definitely an issue. TPS drops to about 0.05 when they attack you

**Comment by mpustovoi** [comment]
Use Spark and its profiler to find the cause of the lag.
Join your world, run `/spark profiler start --timeout 60`, then replicate the issue with a weeping angel and post the link to profiler results here.

**Comment by SteamAutomata** [comment]
I tested in a void world, not only they spawn a lot on the stone plate for some reason, they also cause a massive lag when they are moving. Only one Weeping Angel is enough to drop the tick rate to 0. I tested it on Forge 1.20.1, the mod version is 46.1.5, and I have 167 mods installed, I don't know if the issue is caused by one of those mods.

**Comment by Skullians** [comment]
If the tick rate drops to 0 then the server should do a thread dump eventually. Send it here

**Comment by Jeryn99** [comment]
Should be sorted now with spawn code fixed
